execution_engine/src/core/runtime_context/mod.rs

```diff
@@ -267,6 +267,9 @@ where
                 self.named_keys.remove(name);
                 Ok(())
             }
+            Key::SystemContractRegistry => {
+                unreachable!("should not remove this key once written")
```

Comment:
Should we really be panicking here?
```suggestion
                error!("should not remove the system contract registry key");
                Err(Error::RemoveKeyFailure(RemoveKeyFailure::PermissionDenied))
```

Reply:
Applied suggestion